### PR TITLE
Add song overview list view henry quillin

### PR DESCRIPTION
**Pull request by HenryQuillin**
## What

- Adds song list view on small screens

### Also

- Adds subtle animation to the gradient line above the list view 
- Removes unnecessary prop drilling for history. Adds the useHistory() hook to the components instead. 

## Test

1. Decrease your screen size to less than 900 pixels in width
2. Confirm that playlists are displayed in a list
3. Confirm that clicking appends the songId to history

### CODE REVIEW

**Comment by j-mateo** [comment]
Task linked: [CU-24525km Song Overview List View](https://app.clickup.com/t/24525km)

File: src/components/TableLine.tsx [hunk 1]
```diff
@@ -3,7 +3,7 @@
  */
 
 import { FunctionComponent } from "react";
-import HorizontalLine from "./styled/HorizontalLine";
```
Comment (scandycuz) [inline src/components/TableLine.tsx:6]: I'd like to discuss this animation with the design team. I think we should only be adding animations where we are intentionally trying to draw the user's attention to that element.

**Review by scandycuz — APPROVED** [review]
Looks good to me, just had one comment.

File: src/components/TableLine.tsx (hunk 1 above)
Comment (HenryQuillin) [reply src/components/TableLine.tsx:6]: Ok I agree. Let's discuss with design and change if needed.

Comment (scandycuz) [reply src/components/TableLine.tsx:6]: Sounds good, thanks.

Comment (HenryQuillin) [reply src/components/TableLine.tsx:6]: For the time being I'll make it so slow that its barely noticable

Comment (scandycuz) [reply src/components/TableLine.tsx:6]: Ok

**Pull request merged**